### PR TITLE
Fix accuracy error caused by too many devices in lax_control_flow_test.py

### PR DESCRIPTION
**Pull request by Zantares**
This PR is to limit the devices used in `test_while_loop_of_pmap`, to avoid possible accuracy error when have more than 6 devices. This issue can be reproduced on CPU with below cmd:
```shell
XLA_FLAGS="--xla_force_host_platform_device_count=6"  python tests/lax_control_flow_test.py
```

Error msg:
```
AssertionError:
Not equal to tolerance rtol=1e-06, atol=1e-06

Mismatched elements: 6 / 6 (100%)
Max absolute difference: 2.3841858e-06
Max relative difference: 1.7861877e-06
 x: array([-1.334793, -1.334793, -1.334793, -1.334793, -1.334793, -1.334793],
      dtype=float32)
 y: array([-1.33479, -1.33479, -1.33479, -1.33479, -1.33479, -1.33479],
      dtype=float32)
```

### CODE REVIEW

**Comment by Zantares** [comment]
Hi, can anyone help to review this PR?

**Comment by mattjj** [comment]
@Zantares thanks for this, and thanks for the ping. I'll take a look!

File: tests/lax_control_flow_test.py [hunk 1]
```diff
@@ -2305,20 +2305,27 @@ def arange(n):
...
   @ignore_jit_of_pmap_warning()
   def test_while_loop_of_pmap(self):
-    # code from jsnoek@
+    # Avoid accuracy issue caused by too many devices.
+    DEVICE_LIMITATION = 4
+    devices = jax.devices()
+    count = jax.device_count()
+    if jax.device_count() >= DEVICE_LIMITATION:
+      devices = devices[:DEVICE_LIMITATION]
+      count = DEVICE_LIMITATION
```
Comment (mattjj) [inline tests/lax_control_flow_test.py:2314]: We could probably shorten this, like:

```suggestion
    devices = jax.devices()[:4]
    count = len(devices)
```

but that's not too important, we can merge anyway!

**Review by mattjj — APPROVED** [review]
Thank you!
